lib/hijack/timeout_manager.js

```diff
@@ -0,0 +1,56 @@
+export const TimeoutManager = {
+  id: 0,
+  map: new Map(),
+
+  prettyMessage: {
+    method: (method) => `Method "${method}" still running after`,
+    sub: (sub) => `Subscription "${sub}" still running after`
+  },
+
+  addTimeout (fn, timeout) {
+    if (!fn) {
+      throw new Error('TimeoutManager.addTimeout: fn is required');
+    }
+
+    const id = ++this.id;
+
+    this.map.set(id, setTimeout(Meteor.bindEnvironment(() => {
+      fn();
+
+      this.map.delete(id);
+    }), timeout));
+
+    return id;
+  },
+
+  trackTimeout ({ kadiraInfo, msg, timeout = Kadira.options.stalledTimeout }) {
+    if (!timeout) {
+      return;
+    }
+
+    const type = msg.msg;
+    const method = msg.method || msg.name;
+
+    const error = new Error(`${this.prettyMessage[type](method) || 'Unknown Timeout'} ${timeout}ms`);
```

Comment:
What do you think of showing the timeout in minutes or seconds? With the default of 30 minutes, it will be very large in ms.

Reply:
It would be better I think, I think we can do that easily